[Bug]: Build failure on aarch64 on Mageia Cauldron

### What happened?

```
[ 48%] Building CXX object src/CMakeFiles/conky_core.dir/temphelper.cc.o
cd /home/iurt/rpmbuild/BUILD/conky-1.21.2/build/src && /usr/bin/g++ -D_LARGEFILE64_SOURCE -D_POSIX_C_SOURCE=200809L -Dconky_core_EXPORTS -I/home/iurt/rpmbuild/BUILD/conky-1.21.2/3rdparty/toluapp/include -I/home/iurt/rpmbuild/BUILD/conky-1.21.2/build -I/usr/include/freetype2 -I/usr/include/pango-1.0 -I/usr/include/libmount -I/usr/include/uuid -I/usr/include/json-c -I/usr/include/blkid -I/usr/include/fribidi -I/usr/include/cairo -I/usr/include/libxml2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-6 -I/usr/include/pixman-1 -I/home/iurt/rpmbuild/BUILD/conky-1.21.2/3rdparty/Vc -I/home/iurt/rpmbuild/BUILD/conky-1.21.2/build/src -I/home/iurt/rpmbuild/BUILD/conky-1.21.2/build/data -O2 -g -pipe -Wformat -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -O3 -g -std=c++17 -fPIC -Werror -MD -MT src/CMakeFiles/conky_core.dir/temphelper.cc.o -MF CMakeFiles/conky_core.dir/temphelper.cc.o.d -o CMakeFiles/conky_core.dir/temphelper.cc.o -c /home/iurt/rpmbuild/BUILD/conky-1.21.2/src/temphelper.cc
/home/iurt/rpmbuild/BUILD/conky-1.21.2/src/scroll.cc: In function 'void print_scroll(text_object*, char*, unsigned int)':
/home/iurt/rpmbuild/BUILD/conky-1.21.2/src/scroll.cc:278:10: error: 'char* __builtin___strncat_chk(char*, const char*, long unsigned int, long unsigned int)' specified bound depends on the length of the source argument [-Werror=stringop-overflow=]
  278 |   strncat(pwithcolors, p, pwithcolors_len);
      |          ^
/home/iurt/rpmbuild/BUILD/conky-1.21.2/src/scroll.cc:273:31: note: length computed here
  273 |   int pwithcolors_len = strlen(p) + 4 + colorchanges - visibcolorchanges;
      |                         ~~~~~~^~~
cc1plus: all warnings being treated as errors
gmake[2]: *** [src/CMakeFiles/conky_core.dir/build.make:361: src/CMakeFiles/conky_core.dir/scroll.cc.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
gmake[2]: Leaving directory '/home/iurt/rpmbuild/BUILD/conky-1.21.2/build'
gmake[1]: *** [CMakeFiles/Makefile2:609: src/CMakeFiles/conky_core.dir/all] Error 2
gmake[1]: Leaving directory '/home/iurt/rpmbuild/BUILD/conky-1.21.2/build'
gmake: *** [Makefile:169: all] Error 2

error: Bad exit status from /home/iurt/rpmbuild/tmp/rpm-tmp.MZ8vTy (%build)

RPM build errors:
    Bad exit status from /home/iurt/rpmbuild/tmp/rpm-tmp.MZ8vTy (%build)
I: [iurt_root_command] ERROR: chroot
```

### Version

1.21.2

### Which OS/distro are you seeing the problem on?

Mageia Cauldron aarch64

### Relevant log output

https://pkgsubmit.mageia.org/uploads/failure/cauldron/core/release/20240530170312.kekepower.duvel.2376268/conky-1.21.2-1.mga10/build.aarch64.0.20240530170410.log

--- a/src/scroll.cc
+++ b/src/scroll.cc
@@ -284,7 +284,7 @@ void print_scroll(struct text_object *obj, char *p, unsigned int p_max_size) {
   }
   pwithcolors[strend + j] = 0;
   strncpy(p, pwithcolors, p_max_size);
-  free(pwithcolors);
+  delete[] pwithcolors;
 
   // scroll
   if (sd->direction == SCROLL_LEFT) {

--- a/src/scroll.cc
+++ b/src/scroll.cc
@@ -271,11 +271,11 @@ void print_scroll(struct text_object *obj, char *p, unsigned int p_max_size) {
   }
 
   int pwithcolors_len = strlen(p) + 4 + colorchanges - visibcolorchanges;
-  char *pwithcolors = static_cast<char *>(malloc(pwithcolors_len));
+  char *pwithcolors = new char[pwithcolors_len];
 
   for (j = 0; j < frontcolorchanges; j++) { pwithcolors[j] = SPECIAL_CHAR; }
   pwithcolors[j] = 0;
-  strncat(pwithcolors, p, pwithcolors_len);
+  strcat(pwithcolors, p);
   unsigned int strend = strlen(pwithcolors);
   // and place the colorchanges not in front or in the visible part behind the
   // visible part

